tools/gfx/d3d12/d3d12-device.cpp

```diff
@@ -1341,14 +1349,26 @@ Result DeviceImpl::createTextureView(
             {
                 rtvDesc.ViewDimension = isArray ? D3D12_RTV_DIMENSION_TEXTURE2DARRAY
                     : D3D12_RTV_DIMENSION_TEXTURE2D;
-                rtvDesc.Texture2DArray.MipSlice = desc.subresourceRange.mipLevel;
-                rtvDesc.Texture2DArray.PlaneSlice =
-                    resourceImpl ? D3DUtil::getPlaneSlice(
-                        D3DUtil::getMapFormat(resourceImpl->getDesc()->format),
-                        desc.subresourceRange.aspectMask)
-                    : 0;
-                rtvDesc.Texture2DArray.ArraySize = desc.subresourceRange.layerCount;
-                rtvDesc.Texture2DArray.FirstArraySlice = desc.subresourceRange.baseArrayLayer;
+                if(isArray)
+                {
+                    rtvDesc.Texture2DArray.MipSlice = desc.subresourceRange.mipLevel;
+                    rtvDesc.Texture2DArray.ArraySize = desc.subresourceRange.layerCount;
+                    rtvDesc.Texture2DArray.FirstArraySlice = desc.subresourceRange.baseArrayLayer;
+                    rtvDesc.Texture2DArray.PlaneSlice =
+                        resourceImpl ? D3DUtil::getPlaneSlice(
+                            D3DUtil::getMapFormat(resourceImpl->getDesc()->format),
+                            desc.subresourceRange.aspectMask)
+                        : 0;
+                }
+                else
+                {
+                    rtvDesc.Texture2D.MipSlice = desc.subresourceRange.mipLevel;
+                    rtvDesc.Texture2D.PlaneSlice =
+                        resourceImpl ? D3DUtil::getPlaneSlice(
+                            D3DUtil::getMapFormat(resourceImpl->getDesc()->format),
+                            desc.subresourceRange.aspectMask)
+                        : 0;
```

Comment:
Here, it is important to handle Texture2D.PlaneSlice and Texture2DArray.PlaneSlice differently, since they are at different memory locations in the union.